daemon/protocol/client_lib.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+import os
+import re
+import tempfile
+import shutil
+
+from grpc.beta import implementations
+#import grpc
+import daemon_pb2
+
+
+# Copied from daemon/protocol/locate.go
+def _locate_scoot_dir():
+    scootdir = os.environ.get("SCOOTDIR")
+    if scootdir is not None:
+        return scootdir
+    homedir = os.environ.get("HOME")
+    if homedir is not None:
+        return os.path.join(homedir, ".scoot")
+    return None
+
+def _locate_socket():
+    scootdir = _locate_scoot_dir()
+    if scootdir is None:
+        return None
+    return os.path.join(scootdir, "socket")
+
+
+# Singleton variables.
+_domain_sock = _locate_socket()
+_client = None
+_context = None
+
+class ScootException(Exception):
+    """
+    """
+    def __init__(self, caught):
```

Comment:
Any reason we want any exception class logic at all over:
`class ScootException(Exception):
    pass`
?